commit 1b88c2d086ede9139ace3e0c616dbb23ec55526d
Author: John Levon <john.levon@joyent.com>
Date:   2019-04-03T11:34:52-07:00 (6 months ago)
    
    TRITON-892 Triton documentation updates related to switch to GPT/Loader-based USB keys
    TRITON-1017 iPXE should be provided directly from joyent/ipxe
    TRITON-1158 triton docs build needs fixes on SmartOS

diff --git a/Makefile b/Makefile
index 8545cbc..7375b68 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright (c) 2018, Joyent, Inc.
 #
 
 #
@@ -18,6 +18,7 @@ RAMSEY = node_modules/.bin/ramsey
 MD_FILES = \
 	docs/developer-guide/repos.md
 
+CLEAN_FILES += build/repos.json
 
 
 include ./tools/mk/Makefile.defs
@@ -36,13 +37,13 @@ docs: $(MD_FILES)
 docs/developer-guide/repos.md: $(RAMSEY) docs/developer-guide/repos.md.in build/repos.json
 	$(RAMSEY) -d build -j repos.json "$@.in" "$@"
 
+JSON_SCRIPT = 'var self = this; \
+	this.name = /([^:/]+).git$$/.exec(this.git)[1]; \
+	this.url = "https://github.com/joyent/" + this.name; \
+	(this.tags || ["none"]).forEach(function (t) { self[t] = true; });'
+
 build/repos.json: build etc/repos.json
-	$(JSON) -f ./etc/repos.json -e ' \
-	    var self = this; \
-	    this.name = /([^:/]+).git$$/.exec(this.git)[1]; \
-	    this.url = "https://github.com/joyent/" + this.name; \
-	    (this.tags || ["none"]).forEach(function (t) { self[t] = true; }); \
-	    ' > "$@"
+	$(JSON) -f ./etc/repos.json -e $(JSON_SCRIPT) > "$@"
 
 build:
 	mkdir -p "$@"
diff --git a/docs/developer-guide/coal-setup.md b/docs/developer-guide/coal-setup.md
index 523ec10..1e9e775 100644
--- a/docs/developer-guide/coal-setup.md
+++ b/docs/developer-guide/coal-setup.md
@@ -25,7 +25,7 @@ are not appropriate for production deployments.
 ## Minimum Requirements
 
 The minimum requirements for a good CoaL experience are a Mac with 16 GB RAM and
-an SSD with 45 GB disk available.
+an SSD with 45 GB of disk space available.
 
 ## Overview
 
@@ -87,12 +87,12 @@ To extract the CoaL virtual machine:
 
 ```bash
 $ tar -zxvf coal-latest.tgz -C ~/Documents/Virtual\ Machines.localized
-x root.password.20140911t161518z
-x coal-master-20140911T194415Z-g1a445f5-4gb.vmwarevm/
-x coal-master-20140911T194415Z-g1a445f5-4gb.vmwarevm/USB-headnode.vmx
-x coal-master-20140911T194415Z-g1a445f5-4gb.vmwarevm/zpool.vmdk
-x coal-master-20140911T194415Z-g1a445f5-4gb.vmwarevm/USB-headnode.vmdk
-x coal-master-20140911T194415Z-g1a445f5-4gb.vmwarevm/4gb.img
+x root.password.20190122T132623z
+x coal-master-20190122T132623Z-g1a445f5-4gb.vmwarevm/
+x coal-master-20190122T132623Z-g1a445f5-4gb.vmwarevm/USB-headnode.vmx
+x coal-master-20190122T132623Z-g1a445f5-4gb.vmwarevm/zpool.vmdk
+x coal-master-20190122T132623Z-g1a445f5-4gb.vmwarevm/USB-headnode.vmdk
+x coal-master-20190122T132623Z-g1a445f5-4gb.vmwarevm/4gb.img
 ...
 ```
 
@@ -109,53 +109,42 @@ To configure memory:
 
 1. Select **Virtual Machine** and then click **Settings**.  
    The **USB-headnode Settings** dialog displays.
-1. Click **Processes & Memory**. Change the memory settings, if needed.  
+2. Click **Processes & Memory**. Change the memory settings, if needed.  
    **Note**: The default memory is set to 8192 MB. You can change this setting,
    but be sure to leave Mac OS X with at least 8 GB to avoid resource allocation
    problems. If you are not using the VMware GUI, you can change the default
    memory in the CoaL `USB-headnode.vmx` file. You can set memsize = "6144",
    or according to your preference.
-1. If you have changed the memory settings, restart the VM.
+3. If you have changed the memory settings, restart the VM.
 
-The GRUB boot loader menu displays.
 
-## GRUB menu navigation alert
+## Booting the head node
 
-GRUB intervenes in the boot process, enabling you to specify the environment to
-load. The default is for the console to use ttyb. This is fine for production but
-not for our use.
+1. When you are prompted with the Loader menu press **'5'** to go to the Boot
+Options menu.
 
-Immediately after highlighting **Live 64-bit** using the arrow keys, press **c**
-quickly in order to direct the console to use **vga**.
+     ![CoaL Loader Main Menu](../img/coal-loader-main-menu.png)
 
-![CoaL Grub Boot Menu](../img/coal-grub-menu.png)
+     By default, the OS will redirect the console to ttyb which is fine
+     for production but needs to be changed for CoaL.  Once in the Boot
+     Options menu, press **'2'** to cycle through the choices for **OS Console**
+     until the value is **'text'**, as shown below:
 
-If you are too slow, the **Live 64-bit** option loads instead of the GRUB command
-line. If needed, press **Esc** to redirect the console back to the GRUB menu and
-try again.
+     ![CoaL Loader Boot Options Menu](../img/coal-loader-boot-options.png)
 
-## Booting the head node
+2. Press **'1'** to return to the main menu.
 
-When you are prompted with the GRUB menu:
+3. Press **'2'** to boot from the USB stick.
 
-1. Press the down arrow key to highlight **Live 64-bit**.
-1. Quickly press **c** to load the GRUB command line.  
-   If you miss this step, press **Esc** to redirect the console back to the GRUB
-   menu and try again.
-1. At the **GRUB command prompt**, type `variable os_console vga`, and press **Enter**.
-1. Press **Esc** to return to the GRUB menu.
-1. Press **Enter** to boot **Live 64-bit**.
 
 **Note**: If you see a blank screen with a cursor at the top left while the head
-node is booting, you may have forgotten to redirect the console. If so, press
-`Esc` to return to the GRUB menu.
-
+node is booting, you may have forgotten to redirect the console.
 ![cursor_only](../img/coal-only-cursor.png)
 
 **Note**: Because you are in a virtual environment without Intel VT-x support
-enabled, you'll receive cpu and kvm warnings. You can safely ignore them:
+enabled, you'll receive CPU warnings. You can safely ignore them:
 
-![kvm warning on boot](../img/coal-boot-warnings.png)
+![CPU warning on boot](../img/coal-boot-warnings.png)
 
 ## Configuring the head node
 
@@ -221,15 +210,14 @@ A reboot notification displays when the next phase of installation completes:
 
 ![Reboot message shown on console.](../img/coal-will-reboot.png)
 
-The final phase of installation, setup, is the longest but does not display its
-progress. You may see either just a cursor on the login page or a login prompt.
-
-A message on the console welcomes you to Triton. After some time, a
-**preparing for setup** message displays:
+After the system reboots, images and manifests from the USB key are copied into
+a cache in the zones ZFS pool.  This process can take several minutes:
 
-![preparing for setup on console.](../img/coal-preparing-for-setup.png)
+![Creating the USB cache.](../img/coal-usbkey-cache.png)
 
-When the installation completes, **completing setup** displays.
+The final and longest phase of installation involves importing and provisioning
+the Triton service images.  When the installation completes, **Setup complete**
+displays.
 
 ![Setup complete on console.](../img/coal-setup-complete.png)
 
@@ -436,7 +424,8 @@ minutes depending on how many services have new images.
     config-agent refresh for all the running servers
     Done.
     ```
-1. To update all the services, run the command below. This step can take up to 45 minutes depending on how many services have new images.
+1. To update all the services, run the command below. This step can take up to
+45 minutes depending on how many services have new images.
 
     ```bash
     [root@headnode (coal-1) /var/tmp]# sdcadm update --all -y
diff --git a/docs/developer-guide/repos.md b/docs/developer-guide/repos.md
index 4785546..2fedf72 100644
--- a/docs/developer-guide/repos.md
+++ b/docs/developer-guide/repos.md
@@ -107,10 +107,10 @@ the OS image, and form the foundation for SDC:
 Build tools are used for creating the zone images:
 
 * [sdcnode](https://github.com/joyent/sdcnode): Tools for creation of prebuilt node tarballs for SDC components.
-* [sdcboot](https://github.com/joyent/sdcboot): SDC FDUM environment
 * [sdc-headnode](https://github.com/joyent/sdc-headnode): Repository for building headnode images for SDC, and the intial setup and configuration of the headnode itself
 * [sdc-scripts](https://github.com/joyent/sdc-scripts): Common scripts for configuring and setting up SDC zones.
 * [mountain-gorilla](https://github.com/joyent/mountain-gorilla): Builder of all the SDC bits.
+* [ipxe](https://github.com/joyent/ipxe): SDC iPXE implementation
 
 
 There are also services responsible for syncing data between datacenters or
diff --git a/docs/developer-guide/repos.md.in b/docs/developer-guide/repos.md.in
index 8c2dbed..9e8693d 100644
--- a/docs/developer-guide/repos.md.in
+++ b/docs/developer-guide/repos.md.in
@@ -1,7 +1,7 @@
-# SmartDataCenter repos
+# Triton DataCenter repos
 
 This page holds an overview of the source repositories that make up
-SmartDataCenter (SDC). See also [the reference](../reference.md).
+Triton DataCenter. See also [the reference](../reference.md).
 
 
 The customer-facing self-service API:
diff --git a/docs/img/coal-boot-warnings.png b/docs/img/coal-boot-warnings.png
index 4d6e736..2891488 100644
Binary files a/docs/img/coal-boot-warnings.png and b/docs/img/coal-boot-warnings.png differ
diff --git a/docs/img/coal-grub-menu.png b/docs/img/coal-grub-menu.png
deleted file mode 100644
index 688e5a9..0000000
Binary files a/docs/img/coal-grub-menu.png and /dev/null differ
diff --git a/docs/img/coal-loader-boot-options.png b/docs/img/coal-loader-boot-options.png
new file mode 100644
index 0000000..08ad5db
Binary files /dev/null and b/docs/img/coal-loader-boot-options.png differ
diff --git a/docs/img/coal-loader-main-menu.png b/docs/img/coal-loader-main-menu.png
new file mode 100644
index 0000000..3d4133f
Binary files /dev/null and b/docs/img/coal-loader-main-menu.png differ
diff --git a/docs/img/coal-only-cursor.png b/docs/img/coal-only-cursor.png
index 3166170..5fb9cb5 100644
Binary files a/docs/img/coal-only-cursor.png and b/docs/img/coal-only-cursor.png differ
diff --git a/docs/img/coal-preparing-for-setup.png b/docs/img/coal-preparing-for-setup.png
deleted file mode 100644
index e81cd99..0000000
Binary files a/docs/img/coal-preparing-for-setup.png and /dev/null differ
diff --git a/docs/img/coal-setup-complete.png b/docs/img/coal-setup-complete.png
index b7ec47d..42cf90a 100644
Binary files a/docs/img/coal-setup-complete.png and b/docs/img/coal-setup-complete.png differ
diff --git a/docs/img/coal-usbkey-cache.png b/docs/img/coal-usbkey-cache.png
new file mode 100644
index 0000000..edce23b
Binary files /dev/null and b/docs/img/coal-usbkey-cache.png differ
diff --git a/docs/img/coal-verify-configuration.png b/docs/img/coal-verify-configuration.png
index 0c2cfd7..11d72b0 100644
Binary files a/docs/img/coal-verify-configuration.png and b/docs/img/coal-verify-configuration.png differ
diff --git a/docs/img/coal-will-reboot.png b/docs/img/coal-will-reboot.png
index 1f1daaf..7486f20 100644
Binary files a/docs/img/coal-will-reboot.png and b/docs/img/coal-will-reboot.png differ
diff --git a/etc/repos.json b/etc/repos.json
index 44d6b95..8cf37d8 100644
--- a/etc/repos.json
+++ b/etc/repos.json
@@ -253,13 +253,6 @@
       "cli"
     ]
   },
-  {
-    "git": "git@github.com:joyent/sdcboot.git",
-    "desc": "SDC FDUM environment",
-    "tags": [
-      "build"
-    ]
-  },
   {
     "git": "git@github.com:joyent/sdc-ufds.git",
     "desc": "SDC LDAP Server",
@@ -509,6 +502,13 @@
       "build"
     ]
   },
+  {
+    "git": "git@github.com:joyent/ipxe.git",
+    "desc": "SDC iPXE implementation",
+    "tags": [
+      "build"
+    ]
+  },
   {
     "git": "git@github.com:joyent/eng.git",
     "desc": "Joyent Engineering Guide",
diff --git a/repos.json b/repos.json
index 9b91410..4f7faad 100644
--- a/repos.json
+++ b/repos.json
@@ -83,6 +83,13 @@
         {
             "name": "grafana"
         },
+        {
+            "name": "ipxe",
+            "labels": {
+                "release": true,
+                "mg": "ipxe"
+            }
+        },
         {
             "name": "java-triton"
         },
@@ -1045,7 +1052,6 @@
         "image-converter",
         "imagetools",
         "innovator-workshop",
-        "ipxe",
         "java-fast-md5",
         "java-http-signature",
         "java-manta",
